### PR TITLE
Add `configDescription` field for plugins

### LINKED ISSUE #506: Let plugins specify general config page description
A description that gives context to the config, like links to API docs, etc.

See: https://github.com/wulkano/kap/pull/504#issuecomment-423228417

### PR DESCRIPTION
**Pull request by karaggeorge**
Closes #506 

Waiting for @skllcrn to confirm the design

Also need some suggestions on the wording in the docs

Screen Shot 2019-11-01 at 11 34 13 AM

### CODE REVIEW

File: docs/plugins.md [hunk 1]
```diff
@@ -44,6 +44,7 @@ In the above case, the second and third item are added by two different share se
 The share service is a plain object defining some metadata:
 
 - `title`: The title used in the export menu. For example: `Share to GIPHY`.<br>The text should be in [title case](https://capitalizemytitle.com), for example, `Save to Disk`, not `Save to disk`.
+- `description`: A description that shows up at the top of the configuration window. You can use this to explain the config options or link to API docs. Any links in this description will be parsed into clickable links automatically.
```
Comment (sindresorhus) [inline docs/plugins.md]:
```suggestion
- `description`: A description displayed at the top of the configuration window. You can use this to explain the config options or link to API docs. URLs in the description will automatically be linkified.
```

?

**Comment by sindresorhus** [comment]
Not directly related to this PR, but I feel the config window has too much vertical whitespace. Too much space above the description and between the form elements.

File: docs/plugins.md (hunk 1 above)
Comment (sindresorhus) [reply docs/plugins.md]: Not sure about `description` as a name. That sounds more like it's the description of the plugin, rather than its config.

Comment (karaggeorge) [reply docs/plugins.md]: `configDescription`? It's on the shareService object

**Comment by karaggeorge** [comment]
> Not directly related to this PR, but I feel the config window has too much vertical whitespace. Too much space above the description and between the form elements.

It's set to `padding: 32px 16px`. I agree that there's a lot of vertical space, maybe change to `16px` both sides? Or `24px 16px` 

cc @skllcrn

File: docs/plugins.md (hunk 1 above)
Comment (sindresorhus) [reply docs/plugins.md]: 👍

**Comment by skllcrn** [comment]
I agree, let's try making it quite a bit more compact with 16px on both sides for now?

**Comment by karaggeorge** [comment]
Adjusted the padding to `16px`:

Screenshot 2019-11-12 09 51 46

**Comment by sindresorhus** [comment]
It looks like the description has a bit too much line height.

**Comment by sindresorhus** [comment]
And I would move the input label a tiny bit closer to the input.

**Comment by skllcrn** [comment]
Looks like the description is a smaller font size, which makes the line height a bit too generous. The description should be 14px, and halving the space between the input and label (to 8px) as @sindresorhus mentioned would improve it further.

**Comment by skllcrn** [comment]
Looking really solid buddy

**Comment by karaggeorge** [comment]
Forgot to attach a recent screenshot:

Screenshot 2019-11-13 21 30 11

**Pull request merged**